### PR TITLE
Fix the adapter name for services with java packages

### LINKED ISSUE #1235: WireRpc annotation populated with incorrect adapter locations
In the proto file, you may specify the java_package in addition to the package. For example:
```
package Company.Proto.Auth
option java_package com.example.proto.auth
```
When Wire auto-generates the classes, it will put them under the java_package. For example `com.example.proto.auth.LoginRequest#ADAPTER`.

The auto-generated service will instead use the package, and say something like:
```
  @WireRpc(
    path = "/Company.Proto.Auth.Authentication/Login",
    requestAdapter = "Company.Proto.Auth.LoginRequest#ADAPTER",
    responseAdapter = "Company.Proto.Auth.LoginResponse#ADAPTER"
  )
  suspend fun Login(request: LoginRequest): LoginResponse
```

GrpcMethod.toGrpc then does
```
val requestAdapter = ProtoAdapter.get(wireRpc.requestAdapter) as ProtoAdapter<S>
```
and ProtoAdapter does
```
val hash = adapterString.indexOf('#')
val className = adapterString.substring(0, hash)
val fieldName = adapterString.substring(hash + 1)
return Class.forName(className).getField(fieldName).get(null) as ProtoAdapter<Any>
```

This results in `Class.forName("Company.Proto.Auth.LoginRequest#ADAPTER")` when the actual auto-generated class exists as `com.example.proto.auth.LoginRequest#ADAPTER`; causing ClassNotFoundException.

### PR DESCRIPTION
**Pull request by swankjesse**
Closes #1235

### CODE REVIEW

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 1]
```diff
@@ -162,8 +162,8 @@ class KotlinGenerator private constructor(
     val wireRpcAnnotationSpec = AnnotationSpec.builder(WireRpc::class.asClassName())
         .addMember("path = %S", "/$packageName$serviceName/${rpc.name()}")
         // TODO(oldergod|jwilson) Lets' use Profile for this.
```
Comment (oldergod) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Can remove the TODO I guess?

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt [hunk 2]
```diff
@@ -687,7 +687,7 @@ class KotlinGenerator private constructor(
   }
 
   private fun ProtoType.adapterString() = when {
```
Comment (oldergod) [inline wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Well, that's handy...

File: wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt [hunk 3]
```diff
@@ -276,8 +276,8 @@ class KotlinGeneratorTest {
           |interface RouteGuideBlockingServer : Service {
```
Comment (oldergod) [inline wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt]: Remove the comment line 269

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt (hunk 1 above)
Comment (swankjesse) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: Nope, we still wanna hook that up.

File: wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt (hunk 3 above)
Comment (swankjesse) [reply wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt]: `@WireRpc` _path_ doesn’t use the Java package, but the adapters do!

Comment (oldergod) [reply wire-kotlin-generator/src/test/java/com/squareup/wire/kotlin/KotlinGeneratorTest.kt]: hmmmm

File: wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt (hunk 1 above)
Comment (oldergod) [reply wire-kotlin-generator/src/main/java/com/squareup/wire/kotlin/KotlinGenerator.kt]: hmmmm

**Pull request merged**